File: JeffSebastian/playground/src/main/java/nl/uva/se/ql/evaluation/ValueTable.java
package nl.uva.se.ql.evaluation;

import java.util.HashMap;
import java.util.Map;
import java.util.Map.Entry;

import nl.uva.se.ql.evaluation.value.UndefinedValue;
import nl.uva.se.ql.evaluation.value.Value;

public class ValueTable {

	private Map<String, Value> values;

	public ValueTable() {
		values = new HashMap<String, Value>();
	}

	public Value getValue(String name) {
		if (values.containsKey(name)) {
			return values.get(name);
		}

		return new UndefinedValue();
	}

	public void addValue(String name, Value value) {
		values.put(name, value);
	}

	public boolean containsKey(String name) {
		return values.containsKey(name);
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/QL.Runtime/Evaluation/ValueTable.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.QL.Runtime.Evaluation.Types;

namespace UvA.SoftCon.Questionnaire.QL.Runtime.Evaluation
{
    public class ValueTable
    {
        private IDictionary<string, Value> _valueTable = new Dictionary<string, Value>();

        public void Add(string name, Value value)
        {
            _valueTable.Add(name, value);
        }

        public bool HasValue(string name)
        {
            return _valueTable.ContainsKey(name);
        }

        public Value Get(string name)
        {
            return _valueTable[name];
        }
    }
}
